plugins/es-repository-url/src/test/java/org/elasticsearch/repositories/url/URLRepositoryITest.java

```diff
@@ -29,8 +31,12 @@ public class URLRepositoryITest extends IntegTestCase {
 
     @Override
     protected Settings nodeSettings(int nodeOrdinal) {
+        // Create a relative path
+        File dirPath = TEMPORARY_FOLDER.getRoot().getAbsoluteFile();
+        Path dirName = dirPath.toPath().toAbsolutePath().getFileName();
+        String repoPath = Paths.get(TEMPORARY_FOLDER.getRoot().getAbsolutePath(), "..", dirName.toString()).toString();
```

Comment:
I think strictly speaking this doesn't count as relative path. It is something like `/path/to/.../repo` instead of `../to/repo`
I think to get a proper relative path you'd have to make it relative against `cluster().dataDir` or use something like only `repo/`.

And might be easier to create a `Environment` unit-test instead?